### PR TITLE
Python: Fix missing DictionaryElementContents

### PR DESCRIPTION
**Pull request by RasmusWL**

### CODE REVIEW

**Review by yoff — DISMISSED** [review]
LGTM. I did wonder if we could refactor the code so that the comment about adding new cases would be enforced automatically. I did not see an obvious way, though.

File: python/ql/test/experimental/dataflow/fieldflow/test_dict.py [hunk 1]
```diff
@@ -45,6 +45,16 @@ def test_dict_update():
     SINK(d["key"])  # $ flow="SOURCE, l:-1 -> d['key']"
     SINK(d.get("key"))  # $ flow="SOURCE, l:-2 -> d.get(..)"
 
+
+@expects(2) # $ unresolved_call=expects(..) unresolved_call=expects(..)(..)
```
Comment (yoff) [inline python/ql/test/experimental/dataflow/fieldflow/test_dict.py:49]:
```suggestion
```

Comment (yoff) [reply python/ql/test/experimental/dataflow/fieldflow/test_dict.py:49]: See https://github.com/github/semmle-code/actions/runs/8112809865/job/22174818063#step:26:5473

Comment (RasmusWL) [reply python/ql/test/experimental/dataflow/fieldflow/test_dict.py:49]: thanks :+1: I forgot to run these locally

**Comment by RasmusWL** [comment]
> I did wonder if we could refactor the code so that the comment about adding new cases would be enforced automatically. I did not see an obvious way, though.

I tried to refactor the `moreDictStoreSteps` predicate to use a `string` for the key, so it could be re-used in the `TDictionaryElementContent` "constructor", but that ended up leading to non-monotonic recursion :neutral_face:

**Review by yoff — APPROVED** [review]
LGTM

**Pull request merged**